app/scripts/modules/core/src/entityTag/notifications/EphemeralPopover.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+
+import { HoverablePopover, IEntityTags } from 'core';
+
+import * as moment from 'moment';
+
+export interface IEphemeralPopoverProps {
+  entity?: any;
+}
+
+export interface IEphemeralPopoverState {
+  destroyedFromNow?: string;
+}
+
+export class EphemeralPopover extends React.Component<IEphemeralPopoverProps, IEphemeralPopoverState> {
+  constructor(props: IEphemeralPopoverProps) {
+    super(props);
+    this.state = this.getState(props);
+  }
+
+  private getState(props: IEphemeralPopoverProps): IEphemeralPopoverState {
+    const entityTags: IEntityTags = props.entity.entityTags;
+    if (entityTags) {
+      const ephemeralTag = entityTags.tags.filter(x => x.name === 'spinnaker:ttl')[0];
+      if (ephemeralTag) {
+        return {
+          destroyedFromNow: moment(ephemeralTag.value.expiry).fromNow(true),
```

Comment:
pretty trusting that `expiry` is a legitimate value (`"expiry": "2018-05-15T22:34:42.422Z"`)